src/commands/deploy/deploy.ts

```diff
@@ -70,7 +71,8 @@ export async function deploy(actionContext: IActionContext, arg1?: vscode.Uri |
             } else if (linuxFxVersion.startsWith(LinuxRuntimes.python)) {
                 await promptScmDoBuildDeploy(context, context.effectiveDeployFsPath, LinuxRuntimes.python);
             }
-
+        } else {
+            await fse.writeFile(path.join(context.effectiveDeployFsPath, constants.deploymentFileName), constants.deploymentFile);
```

Comment:
As discussed, the only changes you're doing should be a part of "create", not "deploy". You should be able to leave ".deployment" behavior alone.

Reply:
We won't be able to enable SCM_DO_BUILD_DEPLOY by default at Create as it involves writing a '.deployment' file in the workspace directory and we can't confirm if the directory open in the workspace is the project they want to deploy. 
This is only possible when the "Create Web App..." is run from the workspace folder menu.  
Is there a possibility that I might be missing?

Reply:
`SCM_DO_BUILD_DURING_DEPLOYMENT` as an app setting and `SCM_DO_BUILD_DURING_DEPLOYMENT` in the `.deployment` file do the same thing. You only need one, not both. You're already doing the app setting on create, right? So you should be good

Reply:
I don't know if we should set `SCM_DO_BUILD_DEPLOYMENT` to **true** at Create without deciding on the behavior for deployment. 
The problem comes up at below prompt if a user doesn't select 'Yes' but `SCM_DO_BUILD_DEPLOYMENT` is set **true**. 
![image](https://user-images.githubusercontent.com/17537419/117205849-ff0ab400-ada6-11eb-80c3-0c07f189f73f.png)
In this case, build will happen even when the user selected 'No'. 
One option is to set `SCM_DO_BUILD_DEPLOYMENT` to **false** when user selects 'No' at this prompt. But this wouldn't necessarily solve anything as it is redundant behavior. 

Also, when I am testing this, I have noticed that `SCM_DO_BUILD_DEPLOYMENT` seems to only matter for Linux web apps and not for Windows. Windows deployments are successful even without the `SCM_DO_BUILD_DEPLOYMENT`.

Reply:
That prompt should _**not**_ be shown if the app they're deploying to already has `SCM_DO_BUILD_DEPLOYMENT` as an app setting. The point of that prompt is to encourage people to turn it on and in that case they already have it on.